### PR TITLE
Reimplemented filter chaining API on top of new master

### PR DESCRIPTION
**Pull request by oxtoacart**

### CODE REVIEW

File: http_proxy.go [hunk 1]
```diff
@@ -45,28 +46,16 @@ func main() {
...
-	if err != nil {
-		log.Error(err)
-	}
-
-	// Middleware: Common request filter
-	commonHandler, err := commonfilter.New(httpConnect, testingLocal)
-	if err != nil {
-		log.Error(err)
-	}
+	filterChain := filter.NewChain(
```
Comment (oxtoacart) [inline http_proxy.go]: Here you see how easy it is to set up a filter chain, with filters in the order that they actually get executed.

File: commonfilter/commonfilter.go [hunk 2]
```diff
@@ -1,101 +1,71 @@
...
-		}
-	}
-
+func New(opts *Options) filter.Filter {
 	addrs, err := net.InterfaceAddrs()
 	if err != nil {
 		log.Errorf("Error enumerating local addresses: %v\n", err)
 	}
 
+	localIPs := make([]net.IP, 0, len(addrs))
```
Comment (uaalto) [inline commonfilter/commonfilter.go]: nice detail.

**Comment by uaalto** [comment]
LGTM. Didn't find any potential issue.

**Comment by oxtoacart** [comment]
@uaalto Thanks for the review!

**Pull request merged**